PR: https://github.com/odoo/odoo/pull/118354

From: 90cb44e1e1b6bad9c12f3bd14d1c9e340461e19e
From: Sébastien Theys
Date: 2023-04-21 00:21:53

Structural Changes: 2
Total Changes: 76

[REF] mail, *: rename mail.channel to discuss.channel

* = bus, calendar, crm_livechat, hr, hr_holidays, im_livechat, mail_bot,
    mass_mailing, privacy_lookup, test_discuss_full, test_mail,
    test_mail_full, website_crm_livechat, website_livechat, base

In preparation of splitting discuss and mail modules.

Part of task-3265211

closes odoo/odoo#118354

Related: odoo/upgrade#4553
Related: odoo/enterprise#39661
Signed-off-by: Sébastien Theys (seb) <seb@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/__init__.py
+++ b/addons/hr/models/__init__.py
@@ -14,7 +14,7 @@ from . import hr_plan_activity_type
 from . import hr_plan_employee_activity
 from . import hr_work_location
 from . import mail_alias
-from . import mail_channel
+from . import discuss_channel
 from . import models
 from . import res_config_settings
 from . import res_partner

--- a/addons/hr/models/mail_channel.py
+++ b/addons/hr/models/discuss_channel.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class Channel(models.Model):
-    _inherit = 'mail.channel'
+    _inherit = 'discuss.channel'
 
     subscription_department_ids = fields.Many2many(
         'hr.department', string='HR Departments',

--- a/addons/hr/models/hr_employee.py
+++ b/addons/hr/models/hr_employee.py
@@ -354,7 +354,7 @@ class HrEmployeePrivate(models.Model):
             return employees
         employee_departments = employees.department_id
         if employee_departments:
-            self.env['mail.channel'].sudo().search([
+            self.env['discuss.channel'].sudo().search([
                 ('subscription_department_ids', 'in', employee_departments.ids)
             ])._subscribe_users_automatically()
         onboarding_notes_bodies = {}
@@ -392,7 +392,7 @@ class HrEmployeePrivate(models.Model):
         if vals.get('department_id') or vals.get('user_id'):
             department_id = vals['department_id'] if vals.get('department_id') else self[:1].department_id.id
             # When added to a department or changing user, subscribe to the channels auto-subscribed by department
-            self.env['mail.channel'].sudo().search([
+            self.env['discuss.channel'].sudo().search([
                 ('subscription_department_ids', 'in', department_id)
             ])._subscribe_users_automatically()
         return res
